### ISSUE TITLE
I get the acc 56% by 5 epoch which make me saaaaaaaaaaad

### ISSUE THREAD

**Issue by ruiming46zrm** [TreB1eN/InsightFace_Pytorch#25]
with epoch = 4 ,  batch_size = 256 , number workers = 3 ,  others parameter is the same as author.
and with 4 GPUs training ms1m in my server .
finally , get the accuracy rate = 56% , it's a very sad result. I don't know where I got wrong.  
I don't believe the cause is the such short epoch ~~~
-           -           -              -             -             -            -           -          -           -             - 
464312346@qq.com

**Comment by TreB1eN** [comment]
maybe It's the short epoch, why not train longer?

**Comment by ruiming46zrm** [comment]
> maybe It's the short epoch, why not train longer?
thx for re, however:
Now its running to the 7th epoch ,the results is as following, acc  declined .  I think we cannot get 99% in 20 epoch like author did, might there be any other problems?    thank you 

      results:
      model_2018-12-10-06-42_accuracy:0.5388571428571429_step:77333_None.pth
      model_2018-12-10-07-24_accuracy:0.5381428571428571_step:81882_None.pth
      model_2018-12-10-08-05_accuracy:0.557_step:86431_None.pth
      model_2018-12-10-08-46_accuracy:0.5299999999999999_step:90980_None.pth
      model_2018-12-10-09-27_accuracy:0.514_step:95529_None.pth
      model_2018-12-10-10-08_accuracy:0.5281428571428571_step:100078_None.pth
      model_2018-12-10-10-50_accuracy:0.5449999999999999_step:104627_None.pth
      model_2018-12-10-11-31_accuracy:0.5121428571428571_step:109176_None.pth
      model_2018-12-10-12-12_accuracy:0.6101428571428572_step:113725_None.pth
      model_2018-12-10-12-53_accuracy:0.5367142857142857_step:118274_None.pth
      model_2018-12-10-13-34_accuracy:0.5_step:122823_None.pth
     model_2018-12-10-14-14_accuracy:0.5_step:127372_None.pth
     model_2018-12-10-14-54_accuracy:0.5_step:131921_None.pth
     model_2018-12-10-15-34_accuracy:0.5_step:136470_None.pth

**Comment by TreB1eN** [comment]
What's your lr?

**Comment by ruiming46zrm** [comment]
> What's your lr?

default : 1e-3
I didn't set a new one

**Comment by TreB1eN** [comment]
have you change any code ? because my code dosen't support multi card training

**Comment by ruiming46zrm** [comment]
yes  , in 'Learner.py' :


     def train(self, conf, epochs):
        self.model.train()

        **conf.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
        self.model = nn.DataParallel(self.model,device_ids=[0,1,2,3])
        self.model.to(conf.device)**

        running_loss = 0.
        for e in range(epochs):

**Comment by ruiming46zrm** [comment]
> have you change any code ? because my code doesn't support multi card training

you mean this might influence ?    if so, I plan to have a check  
in fact, I am doubt with the batch_size , when  completing  these 20 epochs, I will take smaller batch_size and try .

**Comment by TreB1eN** [comment]
if you enlarge the batch_size to 4 times, normally you should also increase the lr linearly
